File: bindings/lua/Polycode/SpriteSet.lua
require "Polycode/ResourcePool"

class "SpriteSet" (ResourcePool)


function SpriteSet:SpriteSet(...)
	local arg = {...}
	if type(arg[1]) == "table" and count(arg) == 1 then
		if ""..arg[1].__classname == "ResourcePool" then
			self.__ptr = arg[1].__ptr
			return
		end
	end
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.SpriteSet(unpack(arg))
	end
end

function SpriteSet:setTexture(texture)
	local retVal = Polycode.SpriteSet_setTexture(self.__ptr, texture.__ptr)
end

function SpriteSet:getTexture()
	local retVal =  Polycode.SpriteSet_getTexture(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["shared_ptr<Texture>"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function SpriteSet:addSpriteEntry(newEntry)
	local retVal = Polycode.SpriteSet_addSpriteEntry(self.__ptr, newEntry.__ptr)
end

function SpriteSet:getNumSpriteEntries()
	local retVal =  Polycode.SpriteSet_getNumSpriteEntries(self.__ptr)
	return retVal
end

function SpriteSet:getSpriteEntry(index)
	local retVal = Polycode.SpriteSet_getSpriteEntry(self.__ptr, index)
	if retVal == nil then return nil end
	local __c = _G["shared_ptr<Sprite>"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function SpriteSet:removeSprite(sprite)
	local retVal = Polycode.SpriteSet_removeSprite(self.__ptr, sprite.__ptr)
end

function SpriteSet:loadSpriteSet(fileName)
	local retVal = Polycode.SpriteSet_loadSpriteSet(self.__ptr, fileName)
end

function SpriteSet:addSpriteFrame(frame, assignID)
	local retVal = Polycode.SpriteSet_addSpriteFrame(self.__ptr, frame.__ptr, assignID)
end

function SpriteSet:getNumFrames()
	local retVal =  Polycode.SpriteSet_getNumFrames(self.__ptr)
	return retVal
end

function SpriteSet:getSpriteFrame(index)
	local retVal = Polycode.SpriteSet_getSpriteFrame(self.__ptr, index)
	if retVal == nil then return nil end
	local __c = _G["SpriteFrame"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function SpriteSet:getSpriteFrameByID(frameID)
	local retVal = Polycode.SpriteSet_getSpriteFrameByID(self.__ptr, frameID)
	if retVal == nil then return nil end
	local __c = _G["SpriteFrame"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function SpriteSet:removeFrameByID(frameID)
	local retVal = Polycode.SpriteSet_removeFrameByID(self.__ptr, frameID)
end

function SpriteSet:setSpriteFrame(frame)
	local retVal = Polycode.SpriteSet_setSpriteFrame(self.__ptr, frame.__ptr)
end

function SpriteSet:clearFrames()
	local retVal =  Polycode.SpriteSet_clearFrames(self.__ptr)
end

function SpriteSet:createGridFrames(xCount, yCount, defaultAnchor)
	local retVal = Polycode.SpriteSet_createGridFrames(self.__ptr, xCount, yCount, defaultAnchor.__ptr)
end

function SpriteSet:createFramesFromIslands(minDistance, defaultAnchor)
	local retVal = Polycode.SpriteSet_createFramesFromIslands(self.__ptr, minDistance, defaultAnchor.__ptr)
end

function SpriteSet:getSpriteByName(spriteName)
	local retVal = Polycode.SpriteSet_getSpriteByName(self.__ptr, spriteName)
	if retVal == nil then return nil end
	local __c = _G["shared_ptr<Sprite>"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function SpriteSet:__delete()
	if self then Polycode.delete_SpriteSet(self.__ptr) end
end

File: bindings/javascript/Polycode/SpriteSet.js
function SpriteSet() {
}
Duktape.fin(SpriteSet.prototype, function (x) {
	if (x === SpriteSet.prototype) {
		return;
	}
	Polycode.SpriteSet__delete(x.__ptr)
})

SpriteSet.prototype.setTexture = function(texture) {
	Polycode.SpriteSet_setTexture(this.__ptr, texture)
}

SpriteSet.prototype.getTexture = function() {
	var retVal = new Texture()
	retVal.__ptr = Polycode.SpriteSet_getTexture(this.__ptr)
	return retVal
}

SpriteSet.prototype.loadTexture = function(imageFileName) {
	var retVal = new Texture()
	retVal.__ptr = Polycode.SpriteSet_loadTexture(this.__ptr, imageFileName)
	return retVal
}

SpriteSet.prototype.addSpriteEntry = function(newEntry) {
	Polycode.SpriteSet_addSpriteEntry(this.__ptr, newEntry)
}

SpriteSet.prototype.getNumSpriteEntries = function() {
	return Polycode.SpriteSet_getNumSpriteEntries(this.__ptr)
}

SpriteSet.prototype.getSpriteEntry = function(index) {
	var retVal = new Sprite()
	retVal.__ptr = Polycode.SpriteSet_getSpriteEntry(this.__ptr, index)
	return retVal
}

SpriteSet.prototype.removeSprite = function(sprite) {
	Polycode.SpriteSet_removeSprite(this.__ptr, sprite)
}

SpriteSet.prototype.loadSpriteSet = function(fileName) {
	Polycode.SpriteSet_loadSpriteSet(this.__ptr, fileName)
}

SpriteSet.prototype.addSpriteFrame = function(frame,assignID) {
	Polycode.SpriteSet_addSpriteFrame(this.__ptr, frame,assignID)
}

SpriteSet.prototype.getNumFrames = function() {
	return Polycode.SpriteSet_getNumFrames(this.__ptr)
}

SpriteSet.prototype.getSpriteFrame = function(index) {
	var retVal = new SpriteFrame()
	retVal.__ptr = Polycode.SpriteSet_getSpriteFrame(this.__ptr, index)
	return retVal
}

SpriteSet.prototype.getSpriteFrameByID = function(frameID) {
	var retVal = new SpriteFrame()
	retVal.__ptr = Polycode.SpriteSet_getSpriteFrameByID(this.__ptr, frameID)
	return retVal
}

SpriteSet.prototype.removeFrameByID = function(frameID) {
	Polycode.SpriteSet_removeFrameByID(this.__ptr, frameID)
}

SpriteSet.prototype.setSpriteFrame = function(frame) {
	Polycode.SpriteSet_setSpriteFrame(this.__ptr, frame)
}

SpriteSet.prototype.clearFrames = function() {
	Polycode.SpriteSet_clearFrames(this.__ptr)
}

SpriteSet.prototype.createGridFrames = function(xCount,yCount,defaultAnchor) {
	Polycode.SpriteSet_createGridFrames(this.__ptr, xCount,yCount,defaultAnchor)
}

SpriteSet.prototype.createFramesFromIslands = function(minDistance,defaultAnchor) {
	Polycode.SpriteSet_createFramesFromIslands(this.__ptr, minDistance,defaultAnchor)
}

SpriteSet.prototype.getSpriteByName = function(spriteName) {
	var retVal = new Sprite()
	retVal.__ptr = Polycode.SpriteSet_getSpriteByName(this.__ptr, spriteName)
	return retVal
}
